### PR TITLE
Added cancel_after_first_response test

### PR DESCRIPTION
**Pull request by murgatroid99**

### CODE REVIEW

File: src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m [hunk 1]
```diff
@@ -269,4 +269,37 @@ - (void)testCancelAfterBeginRPC {
   [self waitForExpectationsWithTimeout:1 handler:nil];
 }
 
+- (void)testCancelAfterFirstResponseRPC {
+  __weak XCTestExpectation *expectation = [self expectationWithDescription:@"CancelAfterFirstResponse"];
+  
+  // A buffered pipe to which we never write any value acts as a writer that just hangs.
+  GRXBufferedPipe *requestsBuffer = [[GRXBufferedPipe alloc] init];
+  
+  __block bool receivedResponse = false;
```
Comment (jcanizales) [inline src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m]: s/bool/BOOL, s/false/NO

Comment (murgatroid99) [inline src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m]: done

File: src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m [hunk 2]
```diff
@@ -269,4 +269,37 @@ - (void)testCancelAfterBeginRPC {
   [self waitForExpectationsWithTimeout:1 handler:nil];
 }
 
+- (void)testCancelAfterFirstResponseRPC {
+  __weak XCTestExpectation *expectation = [self expectationWithDescription:@"CancelAfterFirstResponse"];
+  
+  // A buffered pipe to which we never write any value acts as a writer that just hangs.
```
Comment (jcanizales) [inline src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m]: But you _are_ writing some value! :) Instead, you can e.g. add a comment before calling cancel, saying explicitly that we don't tell the requestBuffer that we're done.

Comment (murgatroid99) [inline src/objective-c/examples/Sample/SampleTests/RemoteProtoTests.m]: That was some overzealous copying. I fixed it.

**Comment by jcanizales** [comment]
Is the policy now to wait for Travis, even if we know it shouldn't be affected, to see if there are flaky tests?

**Comment by murgatroid99** [comment]
I think that since we *know* the tests won't be affected by this change, you can cancel the tests and merge.

**Pull request merged**